### PR TITLE
fix: cancel active response for tool call in multimodal agent

### PR DESCRIPTION
**Pull request by longcw**
For https://github.com/livekit/agents/issues/1056

### CODE REVIEW

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py [hunk 1]
```diff
@@ -689,6 +689,41 @@ def __init__(self, sess: RealtimeSession) -> None:
...
+            pending_create_fut = self._sess._response_create_fut
+            if pending_create_fut is not None:
+                if not cancel_existing:
+                    return
+                await pending_create_fut
+
+            pending_resp_id = self._sess._active_response_id
+            if pending_resp_id:
+                if not cancel_existing:
+                    logger.warning(
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:703]: I think there are three behaviors right?
1. canceling existing response, enqueuing new response
2. not canceling existing response, aborting the new response
3. not canceling existing response, enqueuing new response

it seems 1 or 2 are the most common path, but for the sake of API design, should it be:
* on_duplicate: "cancel_existing" | "cancel_new" | "keep_both" ?

Comment (davidzhao) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:703]: not loving the name `acreate` either.. `create` seems so good.. but.. maybe `async request`, like `await response.request()` ?

just an idea. .not sure if I love it either

Comment (longcw) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:703]: Yes, there are three options. I'll update the API. I think `create` is good but it's originally a sync method, not sure if it's ok to change it to an async method which might be a breaking changes as I saw many users are calling `response.create` in the entrypoint.

Comment (davidzhao) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:703]: right, we do not want to break the existing API for that reason.

Comment (longcw) [reply livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py:717]: Modified the `create` to return a `Future` or a `Task`, how do you think?

**Pull request merged**